packages/analytics-core/src/plugins/destination.ts

```diff
@@ -271,55 +252,37 @@ export class Destination implements DestinationPlugin {
     this.config.loggerProvider.warn(getResponseBodyString(res));
 
     this.config.flushQueueSize /= 2;
-
-    const tryable = this.getTryableList(list);
-    this.scheduleTryable(tryable);
+    // All the events are still queued, and will be retried with next trigger
   }
 
   handleRateLimitResponse(res: RateLimitResponse, list: Context[]) {
     const dropUserIds = Object.keys(res.body.exceededDailyQuotaUsers);
     const dropDeviceIds = Object.keys(res.body.exceededDailyQuotaDevices);
-    const throttledIndex = res.body.throttledEvents;
     const dropUserIdsSet = new Set(dropUserIds);
     const dropDeviceIdsSet = new Set(dropDeviceIds);
-    const throttledIndexSet = new Set(throttledIndex);
 
-    const retry = list.filter((context, index) => {
+    const retry = list.filter((context) => {
       if (
         (context.event.user_id && dropUserIdsSet.has(context.event.user_id)) ||
         (context.event.device_id && dropDeviceIdsSet.has(context.event.device_id))
       ) {
         this.fulfillRequest([context], res.statusCode, res.body.error);
         return;
       }
-      if (throttledIndexSet.has(index)) {
-        context.timeout = this.throttleTimeout;
-      }
+
       return true;
     });
 
     if (retry.length > 0) {
       // log intermediate event status before retry
       this.config.loggerProvider.warn(getResponseBodyString(res));
     }
-
-    const tryable = this.getTryableList(retry);
-    this.scheduleTryable(tryable);
-  }
-
-  handleOtherResponse(list: Context[]) {
-    const later = list.map((context) => {
-      context.timeout = context.attempts * this.retryTimeout;
-      return context;
-    });
-
-    const tryable = this.getTryableList(later);
-    this.scheduleTryable(tryable);
+    // All the events are still queued, and will be retried with next trigger
```

Comment:
For rate limiting, shouldn't we reduce the flush queue size?
```
    this.config.flushQueueSize /= 2;
```

Reply:
What do the other SDKs do for rate limiting?

Reply:
old android -> do nothing
new android -> split file, triggerBackOff (increase the flush interval)
new ios -> di nothing
old javascript -> do nothing.

Reply:
I checked the HTTP API [doc](https://www.docs.developers.amplitude.com/analytics/apis/http-v2-api/#429-response-toomanyrequestsfordeviceerror). It says `You should pause sending events for that user or device for a period of 30 seconds before retrying and continue retrying until you no longer receive a 429 response.`

I think only doing one thing, only updating the flushQueueSize or increasing the flush interval is not working.